src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +531,56 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
```

Comment:
Q of logic for is_reboot_pending between here and UbuntuProClient:
Instead of having ubuntu_pro_client_check_success, ubuntu_pro_client_reboot_status, default_exception, what if we preset is_reboot_pending to the default value (i.e. default reboot check) and only overwrite it if ubuntu_pro_prereq_is_met and is installed successfully?

Reply:
As per our discussion, I calculate the default reboot check first. The pro client reboot status is checked next. Everything is logged at the end and pro client status is preferred if available.